src/components/slider/slider.tsx

```diff
@@ -0,0 +1,447 @@
+        this.sliderArea = sliderArea;
+
+        this.props.environment!.addEventListener('mousemove', this.onSliderAreaMouseMove);
+        this.props.environment!.addEventListener('mouseup', this.onSliderAreaMouseUp);
+
+        event.preventDefault();
+        sliderArea.focus();
+
+        const relativeValue = this.getValueFromElementAndPointer(this.sliderArea, event.clientX);
+
+        this.setState({
+            relativeValue,
+            isActive: true
+        });
+
+        this.onDragStart(event.nativeEvent);
+        this.props.onInput!(String(this.getAbsoluteValue(relativeValue)));
+    }
+
+    private onSliderAreaMouseMove(event: MouseEvent) {
+        const relativeValue = this.getValueFromElementAndPointer(this.sliderArea, event.clientX);
+
+        requestAnimationFrame(() => {
+            if (!this.isSliderMounted) {
+                return;
+            }
+
+            this.setState({
+                relativeValue
+            });
+        });
+
+        this.onDrag(event);
+        this.props.onInput!(String(this.getAbsoluteValue(relativeValue)));
+    }
+
+    private onSliderAreaMouseUp(event: MouseEvent) {
+        this.props.environment!.removeEventListener('mousemove', this.onSliderAreaMouseMove);
+        this.props.environment!.removeEventListener('mouseup', this.onSliderAreaMouseUp);
+
+        const relativeValue = this.getValueFromElementAndPointer(this.sliderArea, event.clientX);
+        const value = this.getAbsoluteValue(relativeValue);
+
+        this.setState({
+            relativeValue,
+            isActive: false
+        });
+
+        this.onDragStop(event);
+        this.props.onChange!(value);
+    }
+
+    private onSliderAreaTouchStart(event: React.TouchEvent<HTMLElement>) {
+        if (this.props.disabled) {
+            return;
+        }
+        const sliderArea = event.currentTarget;
+        this.sliderArea = sliderArea;
+
+        this.props.environment!.addEventListener('touchmove', this.onSliderAreaTouchMove);
+        this.props.environment!.addEventListener('touchup', this.onSliderAreaTouchEnd);
+        this.props.environment!.addEventListener('touchend', this.onSliderAreaTouchEnd);
+        this.props.environment!.addEventListener('touchcancel', this.onSliderAreaTouchEnd);
+
+        sliderArea.focus();
+
+        const relativeValue = this.getValueFromElementAndPointer(this.sliderArea, event.touches[0].clientX);
+
+        this.setState({
+            relativeValue,
+            isActive: true
+        });
+
+        event.preventDefault();
+
+        this.onDragStart(event.nativeEvent);
+        this.props.onInput!(String(this.getAbsoluteValue(relativeValue)));
+    }
+
+    private onSliderAreaTouchMove(event: TouchEvent) {
+        const relativeValue = this.getValueFromElementAndPointer(this.sliderArea, event.changedTouches[0].clientX);
+        requestAnimationFrame(() => {
+            if (!this.isSliderMounted) {
+                return;
```

Comment:
you can use return value to destroy request in unmount `https://developer.mozilla.org/en-US/docs/Web/API/Window/cancelAnimationFrame`

the same pattern as `setTimeout`